src/core_plugins/kibana/public/management/sections/objects/lib/get_in_app_url.js

```diff
@@ -0,0 +1,19 @@
+export function getInAppUrl(id, type) {
+  switch (type) {
+    case 'search':
+    case 'searches':
+      return `/discover/${id}`;
+    case 'visualization':
+    case 'visualizations':
+      return `/visualize/edit/${id}`;
+    case 'index-pattern':
+    case 'index-patterns':
+    case 'indexPatterns':
+      return `/management/kibana/indices/${id}`;
+    case 'dashboard':
+    case 'dashboards':
+      return `/dashboard/${id}`;
+    default:
+      return `/${type.toLowerCase()}/${id}`;
```

Comment:
This will cause issues if `type` is `null` or `undefined`. Also true for `id` too, but it won't be an `Error`. Not sure if the code paths that call this method allow that though.